slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -380,8 +392,8 @@ std::string SLAMServiceImpl::GetNextDataFileOffline() {
     if (file_list_offline.size() == 0) {
         file_list_offline = viam::io::ListSortedFilesInDirectory(path_to_data);
     }
-    if (file_list_offline.size() == 0) {
-        throw std::runtime_error("no data in data directory");
+    if (file_list_offline.size() <= 2) {
+        throw std::runtime_error("not enough data in data directory");
```

Comment:
Had to change this cause starting the mapping process with only one datafile produced a garbage map (and sometimes there's a `.DS_Store` file located in the directory so I set the minimum to 3 files)

Reply:
Yes:
1) We should not be deleting files from people's directories. I suspect Macs create `.DS_Store` files for a purpose and we should not mess with that.
2) We should anticipate and handle automatically created files like that within our codebase robustly, which means making sure that we only work with `.pbstream` files directly and ignore everything else is the best course of action. In this piece of code I have to admit though that I'm making an assumption about how likely it is for us to encounter automatically added files - which I'm assuming for now only happens with `.DS_Store` files cause that was the first & only time we ran into this issue. I could write more robust code here that'll actually count the `.pbstream` files but I figured my assumption to be safe enough and it allows us to skip having to count the `.pbstream` files in the directory. If we start seeing more issues like these we can change this code to be more robust, as well.

Reply:
Can we get a comment explaining the 2 then? I think its important to note why for future users